src/shareservice.cpp

```diff
@@ -63,75 +74,93 @@ std::string ShareService::share(const std::vector<std::string> &input, const std
     size_t gTotalBytes = 0;
     size_t gTxBytes = 0;
 
-    for (auto &fp : filePaths){
+    for (auto &fp : filePaths) {
         gTotalBytes += io::Path(fp).getSize();
     }
 
     // Upload
     const int MAX_RETRIES = 10;
 
-    for (auto &fp : filePaths){
+    for (auto &fp : filePaths) {
         io::Path p = io::Path(fp);
-
-        std::string sha256 = Hash::fileSHA256(fp.string());
-        std::string filename = fp.filename().string();
         size_t filesize = p.getSize();
 
-        sfp.filename = filename;
-        sfp.totalBytes = filesize;
-        sfp.txBytes = 0;
-
-        if (p.isAbsolute() && !wd.isParentOf(p.get())){
-            p = p.withoutRoot();
-        }else{
-            p = p.relativeTo(wd.get());
-        }
-
-        LOGD << "Uploading " << p.string();
-
-        int retryNum = 0;
-
-        while(true){
-            try{
-                net::Response res = net::POST(reg.getUrl("/share/upload/" + batchToken))
-                        .multiPartFormData({"file", fp.string()},
-                                           {"path", p.generic()})
-                        .authToken(authToken)
-                        .progressCb([&cb, &files, &sfp, &filesize, &gTotalBytes, &gTxBytes](size_t txBytes, size_t totalBytes){
-                            if (cb == nullptr) return true;
-                            else{
-                                // We cap the txBytes from CURL since it
-                                // includes data transferred from the request
-
-                                // CAUTION: this will require a lock if you use threads
-
-                                gTxBytes -= sfp.txBytes;
-                                sfp.txBytes = std::min(filesize, txBytes);
-                                gTxBytes += sfp.txBytes;
-                                return cb(files, gTxBytes, gTotalBytes);
-                            }
-                        })
-                        //.maximumUploadSpeed(1024*1024)
-                        .send();
-
-                // Token expired?
-                if (res.status() == 401){
-                    LOGD << "Token expired";
-                    authToken = reg.login(ac.username, ac.password);
-                    throw NetException("Unauthorized");
-                }
+        // TODO: Extract methods
+        if (filesize > maxUploadSize) {
+            // Upload in chunks
+            /*
+             * 1) Split file in manageable chunks
```

Comment:
Note: no need to split files, just read them from a specific byte offset. We don't want to write to the file system for this feature.

Reply:
See `mmap` and https://docs.microsoft.com/en-us/windows/win32/memory/creating-a-view-within-a-file

Reply:
Note this might require some research, as I've never done this